cypress/support/commands.ts

```diff
@@ -288,3 +287,160 @@ Cypress.Commands.add('selectRowInDialog', (name: string | RegExp, filter?: boole
+    user_type: 'normal',
+  }).then((user) => user);
+});
+
+Cypress.Commands.add('deleteAwxUser', (user: User) => {
+  if (user.id) {
+    cy.requestDelete(`/api/v2/users/${user.id}/`, true);
+  }
+});
+
+Cypress.Commands.add('createAwxProject', () => {
+  cy.createAwxOrganization().then((organization) => {
+    cy.requestPost<Project>('/api/v2/projects/', {
+      name: 'E2E Project ' + randomString(4),
+      organization: organization.id,
+      scm_type: 'git',
+      scm_url: 'https://github.com/ansible/ansible-tower-samples',
+    }).then((project) => {
+      waitForProjectToFinishSyncing(project.id);
+    });
+  });
+});
+
+Cypress.Commands.add('deleteAwxProject', (project: Project) => {
+  const organizationId = project.organization;
+  // Delete sync job related to project
+  if (project && project.related && typeof project.related.last_job === 'string') {
+    const projectUpdateEndpoint: string = project.related.last_job;
+    cy.requestDelete(projectUpdateEndpoint);
+  }
+  // Delete project
+  cy.requestDelete(`/api/v2/projects/${project.id}/`, true);
+  // Delete organization for the project
+  cy.requestDelete(`/api/v2/organizations/${organizationId.toString()}/`, true);
+});
+
+Cypress.Commands.add('createAwxInventory', () => {
+  cy.createAwxOrganization().then((organization) => {
+    cy.requestPost<Inventory>('/api/v2/inventories/', {
+      name: 'E2E Inventory ' + randomString(4),
+      organization: organization.id,
+    }).then((inventory) => inventory);
+  });
+});
+
+Cypress.Commands.add('deleteAwxInventory', (inventory: Inventory) => {
+  const organizationId = inventory.organization;
+  // Delete organization created for this inventory (this will also delete the inventory)
+  if (organizationId) {
+    cy.requestDelete(`/api/v2/organizations/${organizationId.toString()}/`, true);
+  }
+});
+
+Cypress.Commands.add('createAwxJobTemplate', () => {
+  cy.createAwxProject().then((project) => {
+    cy.createAwxInventory().then((inventory) => {
+      cy.requestPost<JobTemplate>('/api/v2/job_templates/', {
+        name: 'E2E Job Template ' + randomString(4),
+        playbook: 'hello_world.yml',
+        project: project.id.toString(),
+        inventory: inventory.id,
+      }).then((jobTemplate) => jobTemplate);
+    });
+  });
+});
+
+Cypress.Commands.add('deleteAwxJobTemplate', (jobTemplate: JobTemplate) => {
+  const projectId = jobTemplate.project;
+
+  if (jobTemplate.id) {
+    const templateId = typeof jobTemplate.id === 'number' ? jobTemplate.id.toString() : '';
+    cy.requestDelete(`/api/v2/job_templates/${templateId}/`, true);
+  }
+  if (projectId) {
+    cy.requestGet<Project>(`/api/v2/projects/${projectId}/`).then((project) => {
+      // This will take care of deleting the project and the associated org, inventory
+      cy.deleteAwxProject(project);
+    });
+  }
+});
+
+let requestCount = 1;
+
+// Polling to wait till a project is synced
+function waitForProjectToFinishSyncing(projectId: number) {
```

Comment:
I like this approach.  We can build on it/modify it if we need to in the future but I think this is the right approach for right now.